src/utils.ts

```diff
@@ -64,12 +64,24 @@ export function inferJuliaNumThreads(): string {
  * Same as `vscode.commands.registerCommand`, but with added middleware.
  * Currently sends any uncaught errors in the command to crash reporting.
  */
-export function registerCommand(cmd: string, f) {
-    const fWrapped = (...args) => {
+export function registerAsyncCommand<T1 extends unknown[],T2>(cmd: string, f: (...arg: T1) => Promise<T2>) {
```

Comment:
Thoughts on only using one function here which explicitly async-ifies everything?

Reply:
I also added an eslint rule that errors if a function is async when it isn't necessary. I guess we could remove that rule, but on the other hand it does seem good to not introduce unnecessary async functions?